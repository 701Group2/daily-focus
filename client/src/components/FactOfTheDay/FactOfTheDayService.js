function fetchRandomFact(setFact) {
    fetch("https://uselessfacts.jsph.pl/random.json?language=en")
        .then((res) => {
            res.json().then((jsonRes) => {
                if (jsonRes.text) {
                    setFact(jsonRes.text);
                } else {
                    setFact("Sorry, could not get quote");
                }
            });
        })
        .catch((err) => {
            setFact("Sorry, could not get quote");
        });
}

function fetchTodayFact(setFact) {
    fetch("https://uselessfacts.jsph.pl/today.json?language=en")
        .then((res) => {
            res.json().then((jsonRes) => {
                if (jsonRes.text) {
                    setFact(jsonRes.text);
                } else {
                    setFact("Sorry, could not get quote");
                }
            });
        })
        .catch((err) => {
            setFact("Sorry, could not get quote");
        });
}

export { fetchRandomFact, fetchTodayFact };
