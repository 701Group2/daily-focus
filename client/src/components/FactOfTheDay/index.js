import React, { useEffect, useState } from "react";
import Button from "@material-ui/core/Button";

export default function FactOfTheDay(props) {
    const [fact, setFact] = useState("Press button to generate fact");

    async function fetchRandomFact() {
        fetch("https://uselessfacts.jsph.pl/random.json?language=en")
            .then((res) => {
                res.json().then((jsonRes) => {
                    console.log(jsonRes);
                    if (jsonRes.text) {
                        setFact(jsonRes.text);
                    } else {
                        setFact("Sorry, could not get quote");
                    }
                });
            })
            .catch((err) => {
                setFact("Sorry, could not get quote");
            });
    }

    async function fetchTodayFact() {
        fetch("https://uselessfacts.jsph.pl/today.json?language=en")
            .then((res) => {
                res.json().then((jsonRes) => {
                    console.log(jsonRes);
                    if (jsonRes.text) {
                        setFact(jsonRes.text);
                    } else {
                        setFact("Sorry, could not get quote");
                    }
                });
            })
            .catch((err) => {
                setFact("Sorry, could not get quote");
            });
    }

    useEffect(() => {
        fetchTodayFact();
    },[]);
    
    return (
        <div>
            <div> Fact of the Day </div>
            <p> {fact} </p>
            <div>
                <Button onClick={() => fetchRandomFact()}> Random Fact </Button>
            </div>
        </div>
    );
}
