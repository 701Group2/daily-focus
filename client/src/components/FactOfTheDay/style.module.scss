.fact {
    display: flex;
    align-items: center;
    flex-direction: column;
    width: 340px;
    height: fit-content;
    background-color: #ffffff;
    border-radius: 25px;
    box-shadow: 0px 4px 16px 8px rgba(0, 0, 0, 0.1);
}

.buttonContainer {
    display: inline-flex;
    margin-bottom: 25px;
}

.factTitle {
    display: flex;
    align-items: center;
    justify-content: center;
    background: #30a0f5;
    height: 48px;
    width: 100%;
    border-top-left-radius: 25px;
    border-top-right-radius: 25px;
}

.factTitleText {
    color: white;
    font-size: 24px;
    font-family: "Ropa sans";
}

.content {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 10px;
    text-align: center;
}

.contentContainer {
    display: flex;
    text-align: center;
    height: 128px;
}
