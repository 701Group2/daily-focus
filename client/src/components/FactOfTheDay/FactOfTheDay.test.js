import React from "react";
import FactOfTheDay from ".";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";
import Button from "@material-ui/core/Button";

let wrapper;

beforeEach(() => {
    wrapper = shallow(<FactOfTheDay />);
});

test("Fact of the day renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<FactOfTheDay />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("Fact of the Day button renders correctly", () => {
    expect(wrapper.find(Button)).toHaveLength(1);
    expect(wrapper.find(Button).prop("children")).toEqual("Random Fact");
    expect(wrapper.find(Button).props().color).toEqual("primary");
    expect(wrapper.find(Button).props().variant).toEqual("contained");
});

test("Initial content of paragraph rendered correctly", () => {
    const paragraph = '<p class="content">Press button to generate fact</p>';
    expect(wrapper.find("p").html()).toBe(paragraph);
});

test("Component contains Fact Of the Day title", () => {
    expect(wrapper.find(".factTitleText")).toHaveLength(1);
    expect(wrapper.find(".factTitleText").prop("children")).toEqual("Fact of the Day");
});
