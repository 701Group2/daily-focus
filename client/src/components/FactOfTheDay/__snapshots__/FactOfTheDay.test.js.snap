// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fact of the day renders correctly 1`] = `
<div
  className="fact"
>
  <div
    className="factTitle"
  >
    <div
      className="factTitleText"
    >
      Fact of the Day
    </div>
  </div>
  <div
    className="contentContainer"
  >
    <p
      className="content"
    >
      Press button to generate fact
    </p>
  </div>
  <div
    className="buttonContainer"
  >
    <WithStyles(ForwardRef(Button))
      classes={
        Object {
          "label": "makeStyles-label-4",
          "root": "makeStyles-root-3",
        }
      }
      color="primary"
      onClick={[Function]}
      variant="contained"
    >
      Random Fact
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
